### PR TITLE
Introduce the #field meta extractor

### PR DESCRIPTION
**Pull request by tobim**
###  :notebook_with_decorative_cover: Description

This change makes it possible to query vast with expressions of the form
`#field <op> "fieldname"`. The result set will contain all events with layouts
that contain fields named `fieldname` by suffix match.

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [ ] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

By commit.

### CODE REVIEW

File: libvast/src/meta_index.cpp [hunk 1]
```diff
@@ -219,6 +219,31 @@ std::vector<uuid> meta_index::lookup(const expression& expr) const {
...
+            auto s = caf::get_if<std::string>(&d);
+            if (!s) {
+              VAST_WARNING_ANON("#field meta queries only support string "
+                                "comparisons");
+            } else {
+              for (auto& [part_id, part_syn] : synopses_) {
+                bool match = false;
+                for (auto& pair : part_syn.field_synopses_) {
+                  auto fqn = pair.first.fqn();
+                  if (detail::ends_with(fqn, *s)) {
```
[the remark this answers is not in the document]
Comment (tobim) [reply libvast/src/meta_index.cpp:235]: I closer look revealed that `find_suffix` doesn't actually improve the semantics. It's behavior mirrors `ends_with` but it returns a list of offsets, not what we need here.

File: libvast/src/system/exporter.cpp [hunk 2]
```diff
@@ -228,6 +228,7 @@ behavior exporter(stateful_actor<exporter_state>* self, expression expr,
     }
     auto& checker = it->second;
     // Perform candidate check, splitting the slice into subsets if needed.
+    st.query.processed += slice.rows();
```
[the remark this answers is not in the document]
Comment (tobim) [reply libvast/src/system/exporter.cpp:231]: Happens on master too.

[the remark this answers is not in the document]
Comment (tobim) [reply libvast/src/system/exporter.cpp:231]: The reason is the early return on line 236. This change is correct.

**Comment by dominiklohmann** [comment]
This does not currently build with clang: 

```
../libvast/src/meta_index.cpp:235:37: error: reference to local binding 'part_syn' declared in enclosing lambda expression
448
```

File: libvast/src/system/exporter.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (dominiklohmann) [reply libvast/src/system/exporter.cpp:231]: Great catch. Can you write a separate changelog entry for this?

File: libvast/src/meta_index.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (dominiklohmann) [reply libvast/src/meta_index.cpp]: Alright, let's keep it as is for now.

File: libvast/src/taxonomies.cpp [hunk 3]
```diff
@@ -281,92 +281,130 @@ resolve_impl(const taxonomies& ts, const expression& e,
...
+      if (negated(op))
+        expr = negation{std::move(expr)};
+      if (unrestricted.empty())
+        return expr;
+      unrestricted.push_back(expr);
+      return unrestricted;
     };
-    if (auto fe = caf::get_if<field_extractor>(&pred.lhs)) {
-      if (auto data = caf::get_if<vast::data>(&pred.rhs)) {
+    auto abs_op = negated(pred.op) ? negate(pred.op) : pred.op;
```
[the remark this answers is not in the document]
Comment (dominiklohmann) [reply libvast/src/taxonomies.cpp]: I'd still like to see this implemented, but we could also do this as a follow-up to get this merged quicker before the release. Whatever you prefer. If you want to split this change out, please create a story.

**Review by dominiklohmann — APPROVED** [review]
I've run this locally and tested it. Works as expected now.

Do you want me to push the fixups I made for clang?

**Pull request merged**